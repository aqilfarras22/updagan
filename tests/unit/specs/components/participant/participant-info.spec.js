// import sinon from 'sinon';
import BootstrapVue from 'bootstrap-vue';
import { createLocalVue, mount } from '@vue/test-utils';

import participantInfo from '@/components/participants/participant-info';

const { it, describe, expect } = global;

const localVue = createLocalVue();
localVue.use(BootstrapVue);
localVue.use({
  install(vue) {
    vue.filter('moment', (v, format) => format);
  },
});

const studyId = '111';
const participantId = '123';
const testParticipantDetails = {
  id: '123',
};
const getCustomLabelsStub = jest.fn().mockResolvedValue({});
const getStudyDetailsStub = jest.fn().mockResolvedValue({});
const vueAclCheckStub = jest.fn().mockReturnValue(Promise.resolve({}));
const vueAclChangeStub = jest.fn().mockReturnValue(Promise.resolve({}));

const createMountedWrapper = mountOptions => mount(participantInfo, {
  localVue,
  ...mountOptions,
  propsData: {
    participantDetails: testParticipantDetails,
  },
  mocks: {
    $api: {
      getLabelsFact: getCustomLabelsStub,
      getStudyDataFact: getStudyDetailsStub,
    },
    $route: {
      params: {
        studyId,
        participantId,
      },
    },
    $acl: {
      check: vueAclCheckStub,
      change: vueAclChangeStub,
    },
  },
});

describe('components/participants/participant-info.vue', () => {
  it('should create instance', () => {
    const { vm } = createMountedWrapper();
    expect(vm);
  });

  it('should able to receive props', () => {
    const wrapper = createMountedWrapper();
    expect(wrapper.props()).toEqual({
      participantDetails: testParticipantDetails,
    });
  });

  // it.skip('should not create if required props items were not passed', () => {
  //   const spy = jest.spyOn(console, 'error');
  //   mount(participantInfo);
  //   expect(spy).toBeCalledWith(expect.stringContaining('[Vue warn]: Invalid prop'));

  //   spy.mockReset();
  // });

  it('should display props which were passed', () => {
    const wrapper = createMountedWrapper();
    const generatedText = wrapper.text();
    expect(generatedText).toContain(testParticipantDetails.id);
  });

  it('should have method which request all required data: getCustomLabels', async () => {
    // method is called on mount
    expect(getCustomLabelsStub).toBeCalled();

    // getData should change state
    await getCustomLabelsStub();
  });

  it('should have method which request all required data: getStudyDetails', async () => {
    // method is called on mount
    expect(getStudyDetailsStub).toBeCalled();

    // getData should change state
    await getStudyDetailsStub();
  });

  it('has the expected html structure', () => {
    const { vm } = createMountedWrapper();
    expect(vm.$el).toMatchSnapshot();
  });
});
