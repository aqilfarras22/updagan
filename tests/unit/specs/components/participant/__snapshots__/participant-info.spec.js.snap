// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/participants/participant-info.vue has the expected html structure 1`] = `
<div>
  <dl>
    <dt>
      THREAD ID
    </dt>
     
    <dd>
      123
    </dd>
     
    <!---->
  </dl>
   
  <dl>
    <dt>
      Participant ID
      <span
        class="edit-id"
      >
        Edit
      </span>
    </dt>
     
    <dd>
      
    </dd>
  </dl>
   
  <!---->
   
  <!---->
   
  <!---->
   
  <!---->
   
  <!---->
   
  <!---->
    
  <!---->
   
  <dl>
    <!---->
  </dl>
   
  <dl>
    <!---->
  </dl>
   
  <div>
    <div
      id="disqualifyModal___BV_modal_outer_"
    >
      <div
        aria-hidden="true"
        class="modal fade"
        id="disqualifyModal"
        role="dialog"
        style="display: none;"
      >
        <div
          class="modal-dialog modal-lg"
        >
          <div
            aria-describedby="disqualifyModal___BV_modal_body_"
            class="modal-content"
            role="document"
            tabindex="-1"
          >
            <!---->
            <div
              class="modal-body"
              id="disqualifyModal___BV_modal_body_"
            >
              <b>
                <p
                  class="fs-16 pl-10 pr-10"
                >
                  Disqualify participant:
                  <br />
                  
      123?
                </p>
              </b>
               
              <p
                class="pl-10 pr-10"
              >
                For what reason are you disqualifying this participant from this study?
              </p>
               
              <div
                class="pl-10 pr-10"
              >
                <div>
                  <div
                    class="custom-control custom-checkbox custom-control-inline"
                  >
                    <input
                      autocomplete="off"
                      class="custom-control-input"
                      id="__BVID__35"
                      true-value="true"
                      type="checkbox"
                      value="true"
                    />
                    <label
                      class="custom-control-label"
                      for="__BVID__35"
                    >
                      <span
                        class="pl-20"
                      >
                        Participant is not adhering to the protocol-required remote data collection
                      </span>
                    </label>
                  </div>
                  <br />
                </div>
                <div>
                  <div
                    class="custom-control custom-checkbox custom-control-inline"
                  >
                    <input
                      autocomplete="off"
                      class="custom-control-input"
                      id="__BVID__36"
                      true-value="true"
                      type="checkbox"
                      value="true"
                    />
                    <label
                      class="custom-control-label"
                      for="__BVID__36"
                    >
                      <span
                        class="pl-20"
                      >
                        Participant is not attending the protocol-required teleHealth visits
                      </span>
                    </label>
                  </div>
                  <br />
                </div>
                <div>
                  <div
                    class="custom-control custom-checkbox custom-control-inline"
                  >
                    <input
                      autocomplete="off"
                      class="custom-control-input"
                      id="__BVID__37"
                      true-value="true"
                      type="checkbox"
                      value="true"
                    />
                    <label
                      class="custom-control-label"
                      for="__BVID__37"
                    >
                      <span
                        class="pl-20"
                      >
                        Participant is moving and can no longer participate in the study
                      </span>
                    </label>
                  </div>
                  <br />
                </div>
                <div>
                  <div
                    class="custom-control custom-checkbox custom-control-inline"
                  >
                    <input
                      autocomplete="off"
                      class="custom-control-input"
                      id="__BVID__38"
                      true-value="true"
                      type="checkbox"
                      value="true"
                    />
                    <label
                      class="custom-control-label"
                      for="__BVID__38"
                    >
                      <span
                        class="pl-20"
                      >
                        Other
                      </span>
                    </label>
                  </div>
                  <br />
                </div>
                 
                <div
                  class="pl-40 pr-40 pt-10"
                >
                  <!---->
                </div>
              </div>
               
              <div
                class="pb-10 col-10 offset-1"
              >
                <button
                  class="btn mt-20 br-30 btn-primary btn-md btn-block"
                  type="button"
                >
                  Submit
                </button>
              </div>
            </div>
            <!---->
          </div>
        </div>
      </div>
      <!---->
    </div>
  </div>
   
  <div>
    <div
      id="edit-participant-modal___BV_modal_outer_"
    >
      <div
        aria-hidden="true"
        class="modal fade"
        id="edit-participant-modal"
        role="dialog"
        style="display: none;"
      >
        <div
          class="modal-dialog modal-lg"
        >
          <div
            aria-describedby="edit-participant-modal___BV_modal_body_"
            class="modal-content"
            role="document"
            tabindex="-1"
          >
            <!---->
            <div
              class="modal-body"
              id="edit-participant-modal___BV_modal_body_"
            >
              <div
                class="width-400"
              >
                <div>
                  <b>
                    <p
                      class="fs-16 pl-10 pr-10"
                    >
                      Edit participant ID
                      <br />
                    </p>
                  </b>
                </div>
                 
                <div
                  class="pl-10 pr-10"
                >
                  <div
                    class="p-0 col-8"
                  >
                    <input
                      class="form-control"
                      placeholder="Enter new participant ID"
                      type="text"
                    />
                  </div>
                </div>
                 
                <div
                  class="pb-10 pt-40 text-center col-12"
                >
                  <button
                    class="btn mt-20 mr-10 br-3 cancel-btn btn-primary btn-md"
                    type="button"
                  >
                    Cancel
                  </button>
                   
                  <button
                    class="btn mt-20 br-3 save-btn btn-primary btn-md"
                    type="button"
                  >
                    Save
                  </button>
                </div>
              </div>
            </div>
            <!---->
          </div>
        </div>
      </div>
      <!---->
    </div>
  </div>
</div>
`;
